models/sipnet/R/write.configs.SIPNET.R

```diff
@@ -438,7 +446,27 @@ write.config.SIPNET <- function(defaults, trait.values, settings, run.id, inputs
     if ("leafGrowth" %in% pft.names) {
       param[which(param[, 1] == "leafGrowth"), 2] <- pft.traits[which(pft.names == "leafGrowth")]
     }
-  }  ## end loop over PFTS
+
+    #update LeafOnday and LeafOffDay
+    if (settings$model$leaf_phenology){
+     obs_year <- lubridate::year(settings$run$start.date)
+     leaf_pheno_data <- settings$run$inputs$leaf_phenology$path  ## read from settings
+     if (!is.null(leaf_pheno_data)) {
+    ##read data
+       leafphdata <- utils::read.csv(leaf_pheno_data)
+       leafOnDay <- leafphdata$leafonday[leafphdata$year == obs_year & leafphdata$site_id==settings$run$site$id]
+       leafOffDay<- leafphdata$leafoffday[leafphdata$year== obs_year & leafphdata$site_id==settings$run$site$id]
```

Comment:
not required for this PR, but for future reference (1) I suspect dplyr::filter would have been more clear here and (2) it would be good to return a warning to the user if the start.date and end.date are not in the same year, since the current approach doesn't implement any way to update the phenology dates in multi-year runs. Indeed, even in a short term run you could get unexpected results if, for example, you start.date is Dec 31 2023 and your leaf out is in Jan 2024 -- the code as implemented would use the 2023 leaf out date, not the 2024 one.